{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Gene-level RNA-seq differential expression analysis from aligned data\n",
    "[Paolo Sonego](paolo.sonego@fmach.it)<sup>*</sup>\n",
    "\n",
    "<sup>*</sup>Computational Biology Unit,Fondazione Edmund Mach, <br/>\n",
    "Via Edmund Mach 1, 38010, S. Michele All'Adige, Trento,  Italy\n",
    "\n",
    "## Introduction\n",
    "In this practical we will show a standard workflow for the analysis of RNA-seq data. It is not intended as THE Definitive Guide in the analysis of this kind of data; we will cover one of the many existing workflows to analyse RNA-Seq data for a common task, i.e. Differential Expression Analysis.\n",
    "\n",
    "## Analysis overview\n",
    "1. Retrieving RNA-seq reads from public repositories\n",
    "2. Pre-processing (Quality Control, trimming)\n",
    "3. Alignment and Counting\n",
    "4. Exploratory Data Analysis\n",
    "5. Filtering and Normalization\n",
    "6. Differential Expression Analysis\n",
    "7. Gene Ontology Enrichment\n",
    "\n",
    "## Experiment info\n",
    "As a running example we use a dataset which is available from both the Short Read archive (SRA Study ID: SRP193991 - raw data) and from GEO (GEO Experiment ID: GSE130386 - processed data). The authors investigate the Muscat flavour of three Vitis vinifera genotypes with different aromatic strength by integrating transcript and terpene content metabolite (from which the flavour originates) profiles. \n",
    "> A total of 27 fruit samples from three biological replicates were sequenced on Illumina HiSeq2000 at three stages, corresponding to the veraison; berries had intermediate Brix value and were harvest-ripe. \n",
    "\n",
    "For more information about this dataset please refer to the original publication:[Transcriptome profiles of three Muscat table grape cultivars to dissect the mechanism of terpene biosynthesis](https://www.nature.com/articles/s41597-019-0101-y). <a href=\"#ref1\">[1]</a>\n",
    "\n",
    "| Design Type(s) | transcription profiling design • gene expression analysis objective |\n",
    "|--------------------------|---------------------------------------------------------------------|\n",
    "| Measurement Type(s) | transcription profiling assay |\n",
    "| Technology Type(s) | RNA sequencing |\n",
    "| Factor Type(s) | cultivar • biological replicate • developmental stage |\n",
    "| Sample Characteristic(s) | Vitis vinifera • berry |\n",
    "\n",
    "The tools and R packages that we will be using during the practical are listed below and the data used for the analysis can be found at /home/share/RNASeqPractical/.\n",
    "\n",
    "# Setup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Setup - loading libraries for the analysis\n",
    "suppressPackageStartupMessages(library(tidyverse))\n",
    "suppressPackageStartupMessages(library(plotly))\n",
    "suppressPackageStartupMessages(library(RColorBrewer))\n",
    "suppressPackageStartupMessages(library(Rsubread))\n",
    "suppressPackageStartupMessages(library(tximport))\n",
    "suppressPackageStartupMessages(library(DESeq2))\n",
    "suppressPackageStartupMessages(library(edgeR))\n",
    "suppressPackageStartupMessages(library(limma))\n",
    "suppressPackageStartupMessages(library(goseq))\n",
    "suppressPackageStartupMessages(library(GO.db))\n",
    "\n",
    "## Setup - setting up paths and options\n",
    "shared=\"/home/share/RNASeqPractical\"\n",
    "data=\"/home/share/RNASeqPractical/data\"\n",
    "tools=\"/home/share/RNASeqPractical/tools\"\n",
    "references=\"/home/share/RNASeqPractical/references\"\n",
    "options(repr.plot.width = 15, repr.plot.height = 15) # set plot size"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Retrieving RNA-seq reads from public repositories\n",
    "\n",
    "## Download raw data (sra/fastq) from NCBI's Sequence Read Archive (SRA)\n",
    "SRA is one of the primary repositories for next-generation sequencing data. Submissions to SRA can be complex, with multiple samples from a given submission, or multiple sequencing runs associated with a given sample. As a result, a given SRA submission “envelope” can be associated with database records at several levels, each with corresponding identifiers and nomenclature. Valid levels include: the study (prefixed with SRP), the experiment (prefixed with SRX), the sample (prefixed with SRS), and the run (prefixed with SRR). Mapping identifiers between the different parts of an SRA submission is important when you need to access meta-data from different parts of a submission or you need to aggregate data across different levels of a submission (e.g. sequencing runs from the same biological sample)<a href=\"#ref2\">[2]</a>\n",
    "\n",
    "**DO NOT RUN!!!**\n",
    "```{r}\n",
    "suppressPackageStartupMessages(library(SRAdb))\n",
    "suppressPackageStartupMessages(library(DBI))\n",
    "if(!file.exists('SRAmetadb.sqlite')) sra_dbname <<- getSRAdbFile()\n",
    "sra_dbname <- 'SRAmetadb.sqlite'        \n",
    "con <- dbConnect(dbDriver(\"SQLite\"), sra_dbname)\n",
    "listSRAfile('SRP193991', con)\n",
    "getSRAfile('SRP193991', con, fileType = 'fastq', srcType = 'ftp', makeDirectory = TRUE) \n",
    "```\n",
    "or\n",
    "\n",
    "**DO NOT RUN!!!**\n",
    "```sh\n",
    "pysradb download --srp SRP193991\n",
    "```\n",
    "\n",
    "# Quality Control\n",
    "\n",
    "### Activity\n",
    "- Use FastQC  <a href=\"#ref3\">[3]</a> to generate Quality Control reports for the raw data (fastq files).\n",
    "- Check the fastqc report on your browser (notebook session)\n",
    "\n",
    "#### Hint\n",
    "\n",
    "- Open a terminal and browse the RNASeqPractical directory (`/home/share/RNASeqPractical/data`) looking for the fastq raw data\n",
    "- From the command line take advantage of the below template to select few samples to check.\n",
    "\n",
    "```sh\n",
    "cd $HOME/day_2_rnaseq/\n",
    "tools=\"/home/share/RNASeqPractical/tools\"\n",
    "fastq_in=\"<fastq raw path here>\"\n",
    "mkdir -p $HOME/day_2_rnaseq/QC/raw\n",
    "fastqc_out=$HOME/day_2_rnaseq/QC/raw\n",
    "for f in `basename -s .fastq.gz $fastq_in/*.fastq.gz`;\n",
    "do\n",
    "  $tools/FastQC/fastqc $fastq_in/$f.fastq.gz -t 1 --extract -o $fastqc_out\n",
    "done\n",
    "```\n",
    "- Use the below base code to inspect the result on your notebook\n",
    "\n",
    "```{r}\n",
    "qc_list <- list.files(path=\"QC/raw\",pattern=\"*.html\", full.names=T)\n",
    "IRdisplay::display_html(paste0(capture.output(print(paste0(\"<a href=\", qc_list,\">\",qc_list,\"</a>\"), type = 'html')), collapse=\"\"))\n",
    "#IRdisplay::display_html('<iframe src=\"QC/raw/SRR8979775_fastqc/fastqc_report.html\" width=1000, height=1000></iframe> ')\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Trimming\n",
    "Raw data in the fastq format was processed and adaptor and low-quality sequence reads were removed from the data sets using trimmomatic <a href=\"#ref4\">[4]</a>. To run the full samples from this dataset would take several hours.\n",
    "\n",
    "```sh\n",
    "cd $HOME/day_2_rnaseq/\n",
    "tools=\"/home/share/RNASeqPractical/tools\"\n",
    "fastq_in=\"/home/share/RNASeqPractical/data/fastq/raw\"\n",
    "mkdir -p $HOME/day_2_rnaseq/fastq/trimmed\n",
    "fastq_out=$HOME/day_2_rnaseq/fastq/trimmed\n",
    "for f in `basename -s .fastq.gz $fastq_in/*.fastq.gz`;\n",
    "do\n",
    "  java -Xmx2096m -jar $tools/Trimmomatic-0.39/trimmomatic-0.39.jar SE -phred33 -threads 8 \\\n",
    "  $fastq_in/$f.fastq.gz \\\n",
    "  $fastq_out/${f}_trim.fastq.gz \\\n",
    "  ILLUMINACLIP:/home/share/RNASeqPractical/tools/Trimmomatic-0.39/adapters/TruSeq3-PE-2.fa:2:30:10\n",
    "done\n",
    "```\n",
    "\n",
    "### Activity\n",
    "- Open a Terminal, run the trimming software, run FastQC on the result and compare the  report for both pre- and post- trimming fastq files.\n",
    "\n",
    "### hint\n",
    "**This step requires a lot of time so try it with one sample** and check the FastQC reports for both raw and trimmed data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alignment\n",
    "## Build the index\n",
    "The read sequences are stored in (compressed) FASTQ files. These files must be aligned to the Vitis vinifera genome and counted into annotated genes. The first step in performing the alignment is to build an index. In order to build an index you need to have the fasta file (.fa), which can be downloaded from [Vitis vinifera (12x) genome plants ensembl](http://plants.ensembl.org/Vitis_vinifera/Info/Index) or [CRIBI](http://genomes.cribi.unipd.it/DATA) or [INRA](https://urgi.versailles.inra.fr/download/vitis).\n",
    "This operation can be done from either the command line:\n",
    "```sh\n",
    "cd $HOME/day_2_rnaseq/\n",
    "tools=\"/home/share/RNASeqPractical/tools\"\n",
    "references=\"/home/share/RNASeqPractical/references\"\n",
    "mkdir -p cd $HOME/day_2_rnaseq/subread_idx\n",
    "$tools/subread-2.0.0-Linux-x86_64/bin/subread-buildindex -B -o subread_idx/Vitis_vinifera.12X.dna.toplevel.subread $references/ensemblgenomes_Vitis_vinifera_release-43/fasta/dna/Vitis_vinifera.12X.dna.toplevel.fa\n",
    "```\n",
    "or in R using the `Rsubread` package:\n",
    "```{r}\n",
    "references=\"/home/share/RNASeqPractical/references\"\n",
    "dir.create(\"subread_idx\", showWarnings = T)\n",
    "buildindex(basename=\"subread_idx/Vitis_vinifera.12X.dna.toplevel.subread\",\n",
    "reference=paste0(references,\"/ensemblgenomes_Vitis_vinifera_release-43/fasta/dna/Vitis_vinifera.12X.dna.toplevel.fa\"))\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Aligning reads to reference genome\n",
    "\n",
    "Now that we have generated our index, we can align our reads using the align command using the default mapping parameters. To run the full samples from this dataset would take several hours. The BAM files are saved in the working directory.\n",
    "To see how many parameters you can change try the `args` function:```{r} args(align) ``` or better take a look at the [documentation](http://subread.sourceforge.net/).The alignement can be done from either the command line:\n",
    "\n",
    "```sh\n",
    "cd $HOME/day_2_rnaseq/\n",
    "subread_idx=\"$HOME/day_2_rnaseq/subread_idx\"\n",
    "tools=\"/home/share/RNASeqPractical/tools/subread-2.0.0-Linux-x86_64/bin\"\n",
    "fastq=\"/home/share/RNASeqPractical/data/fastq/raw\"\n",
    "#fastq=\"/home/share/RNASeqPractical/data/fastq/trimmed\"\n",
    "mkdir -p $HOME/day_2_rnaseq/bam/raw\n",
    "#mkdir -p $HOME/day_2_rnaseq/bam/trimmed\n",
    "bam=$HOME/day_2_rnaseq/bam/raw\n",
    "#$HOME/day_2_rnaseq/bam/trimmed\n",
    "$tools/subread-align -T 1 -t 0 -i $subread_idx/Vitis_vinifera.12X.dna.toplevel.subread \\\n",
    "  -r $fastq/SRR8979775.fastq.gz -o $bam/SRR8979775.bam\n",
    "```\n",
    "\n",
    "or R:\n",
    "\n",
    "```{r}\n",
    "subread_idx='subread_idx'\n",
    "fastq=\"/home/share/RNASeqPractical/data/fastq/raw\"\n",
    "dir.create('bam/raw', recursive = TRUE)\n",
    "bam=\"bam/raw\"\n",
    "files = list.files(fastq, pattern=\"fastq.gz\",full.names=FALSE)\n",
    "f = 1\n",
    "align(index=paste0(subread_idx,\"/Vitis_vinifera.12X.dna.toplevel.subread\"), \n",
    "      readfile1=list.files(fastq, pattern=\"fastq.gz\",full.names=TRUE)[f], \n",
    "      output_file=paste0(bam, \"/\", tools::file_path_sans_ext(files[f], compression=T),\".bam\"), \n",
    "      nthreads = 1)\n",
    "```\n",
    "\n",
    "### Activity\n",
    "Perform the alignement for the same sample for both raw and trimmed data and check the stats."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Counting\n",
    "Now that we have figured out where each read comes from in the genome, we need to summarise the information across genes or exons. The alignment produces a set of BAM files, where each file contains the read alignments for each library. In the BAM file, there is a chromosomal location for every read that mapped uniquely. The mapped reads can be counted across grape genes by using the featureCounts <a href=\"#ref6\">[6]</a> function from package `RSubread`. The code below uses the exon intervals defined in the NCBI gtf annotation the V1 grape genome. Reads that map to exons of genes are added together to obtain the count for each gene, with some care taken with reads that span exon-exon boundaries. featureCounts takes all the BAM files as input, and outputs an object which includes the count matrix. Each sample is a separate column, each row is a gene.\n",
    "\n",
    "```{r}\n",
    "gtf=\"/home/share/RNASeqPractical/references/ensemblgenomes_Vitis_vinifera_release-43/gtf/Vitis_vinifera.12X.43.chr.gtf\"\n",
    "bam <- \"bam/raw\"\n",
    "bamFiles <- list.files(bam,pattern=\".*bam$\", full.names = TRUE)\n",
    "fc <- featureCounts(files = bamFiles, \n",
    "    nthreads = 1,\n",
    "    strandSpecific = 0,\n",
    "    isPairedEnd = FALSE,\n",
    "    GTF.featureType=\"gene\",\n",
    "    GTF.attrType=\"gene_id\",\n",
    "    annot.ext=gtf, \n",
    "    isGTFAnnotationFile=TRUE )\n",
    "```\n",
    "\n",
    "### Activity\n",
    "\n",
    "- Take a look at the different slots of the featureCounts object (hint: names(fc))\n",
    "- Check the alignment stats.\n",
    "- Save the counts in a text file and write a external representation of the object in a file (see `?save`)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## EDA - Activity\n",
    "- load the full counts matrix from `/home/share/RNASeqPractical/data/counts/subread_counts`; you can select either the trimmed fastq (`fc_trimmed.rda`) or the raw fastq (`fc_raw.rda`) for the forthcoming downstream analysis.\n",
    "- From the counts matrix fc:\n",
    "    - check the library size for all the samples \n",
    "    - show a barplot with library size information\n",
    "    - do a pairwise comparison of samples using scatterplots (or the `smoothScatter()` function)\n",
    "\n",
    "## EDA - barplot\n",
    "```{r}\n",
    "load(\"/home/share/RNASeqPractical/data/counts/subread_counts/fc_raw.rda\")\n",
    "#load(\"/home/share/RNASeqPractical/data/counts/subread_counts/fc_trimmed.rda\")\n",
    "# base R barplot\n",
    "#barplot(colSums(fc$counts), las=2, border = 0, cex.lab=1, cex.axis=1, font=1,col.axis=\"black\")\n",
    "## ggplot2 barplot\n",
    "options(repr.plot.width = 15, repr.plot.height = 15)\n",
    "fc$counts %>% \n",
    "colSums() %>% \n",
    "tibble::enframe(name = NULL) %>% \n",
    "ggplot(aes(x=colnames(fc$counts),y=value)) + \n",
    "geom_col() + \n",
    "theme(axis.text.x = element_text(angle = 90, hjust = 1)) + \n",
    "xlab(\"Samples\") + ylab(\"Successfully assigned alignments\") + coord_flip()  + \n",
    "geom_label(aes(label=scales::comma(colSums(fc$counts))), nudge_y = 30000, size=2.5)\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Design\n",
    "Retrieve metadata about the esperiment and samples. There are several ways to do this using R (e.g. package `GREP2`), the command line (`pysradb`) or directly from the web. My suggestion for this case is to go to [SRA Run Selector](https://www.ncbi.nlm.nih.gov/Traces/study/) and use either the SRA experiment id `SRP193991` or the Bioproject id `PRJNA539964` for downloading `RunInfo Table`; then you can read the downloaded file into R.\n",
    "\n",
    "```{r}\n",
    "SRP193991_metadata <- read.table(\"/home/share/RNASeqPractical/data/SraRunTable.txt\", header=T, sep=\",\")\n",
    "```\n",
    "\n",
    "### Question\n",
    "- Inspect the file\n",
    "- What metadata tell us about the experiment?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Build the target matrix\n",
    "```{r}\n",
    "targets <- SRP193991_metadata[,c(\"Run\",\"sample_acc\", \"GEO_Accession\", \"Cultivar\",\"developmental_stage\",\"replicate\")]\n",
    "colnames(fc$counts) <- str_remove(colnames(fc$counts),\".trim.bam\")\n",
    "```\n",
    "\n",
    "# Sum over replicate samples\n",
    "Before proceeding we need to condense the columns of the counts table so that counts are summed over technical replicate samples.\n",
    "\n",
    "```{r}\n",
    "dge <- edgeR::DGEList(fc$counts)\n",
    "#dge <- edgeR::sumTechReps(dge, ID = targets$sample_acc)\n",
    "dge <- edgeR::sumTechReps(dge, ID = targets$GEO_Accession)\n",
    "targets <- targets[!duplicated(targets$GEO_Accession),]\n",
    "```\n",
    "### Activity\n",
    "Check both the counts matrix and the targets dimensions before and after the summarization."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# EDA \n",
    "\n",
    "## PCA - 2D\n",
    "The RNA samples can be clustered in two dimensions using Principal Component Analysis (PCA) plots <a href=\"#ref7\">[7]</a> showing the first two principal components that explain the variability in the data using the regularized log count data.. This is both an analysis step and a quality control step to explore the overall differences between the expression profiles of the different samples.\n",
    "```{r}\n",
    "dds <- DESeqDataSetFromMatrix(\n",
    "  countData = dge$counts,\n",
    "  colData = targets,\n",
    "  design = ~developmental_stage)\n",
    "vsd <- vst(dds, blind=TRUE)\n",
    "pcaData <- plotPCA(vsd, \n",
    "\tintgroup=c(\"Cultivar\",\"developmental_stage\"), \n",
    "\treturnData=TRUE)\n",
    "percentVar <- round(100 * attr(pcaData, \"percentVar\"))\n",
    "ggplot(pcaData, aes(PC1, PC2, color=Cultivar, shape=developmental_stage)) +\n",
    "  geom_point(size=4) +\n",
    "  xlab(paste0(\"PC1: \",percentVar[1],\"% variance\")) +\n",
    "  ylab(paste0(\"PC2: \",percentVar[2],\"% variance\")) + \n",
    "  coord_fixed() + ggtitle(\"vsd transformed counts: Cultivar and Developmental stage\") +\n",
    "  theme_bw()\n",
    "\n",
    "```\n",
    "## PCA - 3D\n",
    "```{r}\n",
    "tmp <- btools::plotPCA3D(vsd, intgroup = \"developmental_stage\", ntop = 500,  returnData = TRUE)\n",
    "percentVar3D <- round(100 * attr(tmp, \"percentVar\"))\n",
    "plot_ly(tmp, x = ~PC1, y = ~PC2, z = ~PC3, color = ~developmental_stage) %>%\n",
    "  add_markers() %>%\n",
    "  layout(scene = list(xaxis = list(title = paste0(\"PC1: \",percentVar3D[1],\"% variance\")),\n",
    "                      yaxis = list(title = paste0(\"PC2: \",percentVar3D[2],\"% variance\")),\n",
    "                      zaxis = list(title = paste0(\"PC3: \",percentVar3D[3],\"% variance\"))))\n",
    "```\n",
    "### Activity\n",
    "\n",
    "Generate a PCA 3D showing point colored by Cultivar (hint: modify the `intgroup` parameter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Differential Expression Analysis\n",
    "\n",
    "## Design Overview\n",
    "\n",
    "```{r}\n",
    "targets\n",
    "```\n",
    "- 3 cutivars (2 chinese and 1 italian)\n",
    "- 3 developmental stages from \n",
    "\t- Veraison:\n",
    "\t\t+ El-35: Berries begin to colour and enlarge\n",
    "\t\t+ EL-36: Berries with intermediate sugar values\n",
    "\t- Harvest\n",
    "\t\t+ EL-38: Berries havest-ripe\n",
    "\n",
    "## Comparisons (Exercise)\n",
    "\n",
    "Q: Which biological questions do you want to answer with this design?\n",
    "\n",
    "### A. Between developmental_stage, same Cultivar\n",
    "\n",
    "1. X-EL35vsX-EL36\n",
    "2. X-EL35vsX-EL38\n",
    "3. X-EL36vsX-EL38\n",
    "4. Y-EL35vsy-EL36\n",
    "5. Y-EL35vsy-EL38\n",
    "6. Y-EL36vsy-EL38\n",
    "7. Z-EL35vsZ-EL36\n",
    "8. Z-EL35vsZ-EL38\n",
    "9. Z-EL36vsZ-EL38\n",
    "\n",
    "### B. Between Cultivar, same developmental_stage\n",
    "\n",
    "1. X-EL35vsY-EL35\n",
    "2. X-EL35vsZ-EL35\n",
    "3. Y-EL35vsZ-EL35\n",
    "4. X-EL36vsY-EL36 \n",
    "5. X-EL36vsZ-EL36\n",
    "6. Y-EL36vsZ-EL36\n",
    "7. X-EL38vsY-EL38\n",
    "8. X-EL38vsZ-EL38\n",
    "9. Y-EL38vsZ-EL38\n",
    "\n",
    "\n",
    "__All samples will be filtered and normalized together.__\n",
    "\n",
    "\n",
    "## Design Group-Mean Parametrization\n",
    "\n",
    "Linear modeling and differential expression analysis in limma requires a design matrix to be specified. The design matrix records which treatment conditions were applied to each samples, and it also defines how the experimental effects are parametrized in the linear models. The experimental design for this study can be viewed as a one-way layout and the design matrix can be constructed by:\n",
    "\n",
    "```{r}\n",
    "targets$group <- paste0(targets$Cultivar, \".\", targets$developmental_stage)\n",
    "f <-factor(targets$group,levels=unique(targets$group))\n",
    "design <- model.matrix(~0+f)\n",
    "colnames(design) <- unique(targets$group)\n",
    "```\n",
    "\n",
    "# Filtering\n",
    "Genes that have very low counts across all the libraries should be removed prior to downstream analysis. This is justified on both biological and statistical grounds. From biological point of view, a gene must be expressed at some minimal level before it is likely to be translated into a protein or to be considered biologically important. From a statistical point of view, genes with consistently low counts are very unlikely be assessed as significantly DE because low counts do not provide enough statistical evidence for a reliable judgement to be made. Such genes can therefore be removed from the analysis without any loss of information. As a rule of thumb, we require that a gene have a count of at least 10 or so in at least some libraries before it is considered to be expressed in the study. In practice, the filtering is actually based on count-per-million (CPM) values so as to avoid favoring genes that are expressed in larger libraries over those expressed in smaller libraries. In edgeR, the filtering can be accomplished using a function that takes into account the library sizes and the experimental design:\n",
    "```{r}\n",
    "# The next step is to remove rows that consistently have zero or very low counts.\n",
    "keep <- filterByExpr(dge, design)\n",
    "table(keep)\n",
    "dge <- dge[keep,]\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Normalization for composition bias\n",
    "\n",
    "Normalization by trimmed mean of M values (TMM) is performed by using the calcNormFactors function, which returns the DGEList argument with only the norm.factors changed. It calculates a set of normalization factors, one for each sample, to eliminate composition biases between libraries. The product of these factors and the library sizes defines the effective library size, which replaces the original library size in all downstream analyses. The normalization factors of all the libraries multiply to unity. A normalization factor below one indicates that a small number of high count genes are monopolizing the sequencing, causing the counts for other genes to be lower than would be usual given the library size. As a result, the effective library size will be scaled down for that sample. Here we see, for example, tha GSM3737302, GSM3737303, GSM3737308 samples (there are more) have low normalization factors. This is a sign that these samples contain a number of very highly upregulated genes. Note. In general, we find TMM normalization to be satisfactory for almost all well-designed mRNA gene expression experiments. Single-cell RNA-seq is an exception, for which specialized normalization methods are needed (Lun, Bach, and Marioni 2016). Another, less common, type of study requiring special treatment is that with global differential expression, with more than half of the genome differentially expressed between experimental conditions in the same direction (Wu et al. 2013). Global differential expression should generally be avoided in well designed experiments. When it can’t be avoided, then some normalization reference such as spike-ins needs to be built into the experiment for reliable normalization to be done (Risso et al. 2014).\n",
    "\n",
    "```{r}\n",
    "dge <- calcNormFactors(dge, method = \"TMM\")\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Defining each treatment combination as a group\n",
    "We now consider experiments with more than one experimental factor, but in which every\n",
    "combination of experiment conditions can potentially have a unique effect. For example,\n",
    "suppose that an experiment has been conducted on 3 different cultivars at three\n",
    "developmental stages with all samples obtained from independent subjects. \n",
    "The data frame targets describes the treatment conditions applied to each sample.\n",
    "Here we provide a contrast matrix for the contrast `development stage`:\n",
    "\n",
    "## Contrast Matrices\n",
    "```{r}\n",
    "cont.matrix.EL <- makeContrasts(\n",
    "  Xiangfei.EL35vsXiangfei.EL36=Xiangfei.EL35-Xiangfei.EL36, \n",
    "  Xiangfei.EL35vsXiangfei.EL38=Xiangfei.EL35-Xiangfei.EL38, \n",
    "  Xiangfei.EL36vsXiangfei.EL38=Xiangfei.EL36-Xiangfei.EL38, \n",
    "  Italia.EL35vsItalia.EL36=Italia.EL35-Italia.EL36,\n",
    "  Italia.EL35vsItalia.EL38=Italia.EL35-Italia.EL38,\n",
    "  Italia.EL36vsItalia.EL38=Italia.EL36-Italia.EL38,\n",
    "  Zaomeiguixiang.EL35vsZaomeiguixiang.EL36=Zaomeiguixiang.EL35-Zaomeiguixiang.EL36,\n",
    "  Zaomeiguixiang.EL35vsZaomeiguixiang.EL38=Zaomeiguixiang.EL35-Zaomeiguixiang.EL38,\n",
    "  Zaomeiguixiang.EL36vsZaomeiguixiang.EL38=Zaomeiguixiang.EL36-Zaomeiguixiang.EL38,\n",
    "  levels=design)\n",
    "```\n",
    "# Differential expression\n",
    "\n",
    "## Linear modeling with limma-voom\n",
    "\n",
    "We se voom() <a href=\"#ref10\">[10]</a> to convert the read counts to log2-cpm, with associated weights, ready for linear modelling. The voom plot is used to check the mean-variance relationship of the expression data, after fitting a linear model. A scatterplot of residual-variances vs average log-expression is created. The plot is especially useful for examining the mean-variance trend.  It can be seen from this plot that the variance is no longer dependent on the mean expression level.\n",
    "```{r}\n",
    "cont.matrix <- cont.matrix.EL\n",
    "v <- voom(dge, design, plot = TRUE)\n",
    "fit_voom <- lmFit(v, design)\n",
    "plotSA(fit_voom)\n",
    "abline(a=1, b=0, col=\"blue\")\n",
    "fit_voom2 <- contrasts.fit(fit_voom, cont.matrix)\n",
    "fit_voom2 <- eBayes(fit_voom2)\n",
    "fit_voom2$Genes <- rownames(fc$counts)\n",
    "```\n",
    "## Developmental stages (EL) contrasts\n",
    "```{r}\n",
    "Xiangfei.EL35vsXiangfei.EL36 <- topTable(fit_voom2, coef=1,number = Inf, adjust.method = \"BH\", sort.by = \"P\")\n",
    "Xiangfei.EL35vsXiangfei.EL36 %>% rownames_to_column() %>% arrange(-logFC,P.Value) %>% head()\n",
    "fit_voom2_ELs_decideTest_separate <- summary(decideTests(fit_voom2, method=\"separate\"))\n",
    "nice.col <- brewer.pal(6,name=\"Dark2\")\n",
    "stripchart(v$E[\"VIT_07s0005g03420\",]~targets$group,\n",
    "\tvertical=TRUE,las=2,cex.axis=0.8,pch=16,cex=1.3,col=nice.col,\n",
    "\tmethod=\"jitter\",ylab=\"Normalised log2 expression\",main=\"VIT_07s0005g03420\")\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cultivar contrasts - voom\n",
    "```{r}\n",
    "cont.matrix <- cont.matrix.cultivar\n",
    "fit_voom2 <- contrasts.fit(fit_voom, cont.matrix)\n",
    "fit_voom2 <- eBayes(fit_voom2)\n",
    "fit_voom2$Genes <- rownames(fc$counts)\n",
    "Xiangfei.EL35vsItalia.EL35 <- topTable(fit_voom2, coef=1,number = Inf, adjust.method = \"BH\", sort.by = \"P\")\n",
    "Xiangfei.EL35vsItalia.EL35 %>% rownames_to_column() %>% arrange(-logFC,P.Value) %>% head()\n",
    "fit_voom2_cutltivars_decideTest_separate <- summary(decideTests(fit_voom2, method=\"separate\"))\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Differential expression: limma-trend\n",
    "If the sequencing depth is reasonably consistent across the RNA samples, then the simplest and most\n",
    "robust approach to differential exis to use limma-trend. This approach will usually work well if the\n",
    "ratio of the largest library size to the smallest is not more than about 3-fold.\n",
    "In the limma-trend approach, the counts are converted to logCPM values using edgeR’s cpm\n",
    "function:\n",
    "```{r}\n",
    "logCPM <- cpm(dge, log=TRUE, prior.count=3)\n",
    "```\n",
    "__N.B.__ The prior count is used here to damp down the variances of logarithms of low counts.\n",
    "\n",
    "The logCPM values can then be used in any standard limma pipeline, using the trend=TRUE argument when running eBayes or treat."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Activity\n",
    "- Create a contrast matrix ( named `cont.matrix.cultivar` ) which allows the selection of the differential expressed genes for all three cultivars in any stage, i.e.Xiangfei.EL35vsItalia.EL35, Xiangfei.EL35-Italia.EL35, Xiangfei.EL35vsZaomeiguixiang.EL35, etc..\n",
    "- Use either the limma-voom or the following limma-trend procedure to retrieve a list of differential expressed genes for each contrast."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cultivar contrasts - trend \n",
    "```{r}\n",
    "cont.matrix <- cont.matrix.cultivar\n",
    "fit_voom <- lmFit(logCPM, design)\n",
    "fit_voom2 <- contrasts.fit(fit_voom, cont.matrix)\n",
    "fit_voom2 <- eBayes(fit_voom2, trend = TRUE)\n",
    "fit_voom2$Genes <- rownames(fc$counts)\n",
    "Xiangfei.EL35vsItalia.EL35 <- topTable(fit_voom2, coef=1,number = Inf, adjust.method = \"BH\", sort.by = \"P\")\n",
    "Xiangfei.EL35vsItalia.EL35 %>% rownames_to_column() %>% arrange(-logFC,P.Value) %>% write_tsv(\"tT_Xiangfei.EL35vsItalia.EL35_trend.txt\")\n",
    "fit_voom2_cutltivars_decideTest_separate <- summary(decideTests(fit_voom2, method=\"separate\"))\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### plot of individual genes of interest\n",
    "Before following up on the DE genes with further lab work, it is recommended to have a look at the expression levels of the individual samples for the genes of interest. We can quickly look at grouped expression using stripchart. We can use either the normalised log expression values in the voom object (`v$E`) or `logCPM`.\n",
    "\n",
    "```{r}\n",
    "nice.col <- brewer.pal(6,name=\"Dark2\")\n",
    "my_gene=\"VIT_11s0052g01650\"\n",
    "stripchart(v$E[my_gene,]~targets$group,\n",
    "  vertical=TRUE,las=2,cex.axis=0.8,pch=16,cex=1.3,col=nice.col,\n",
    "  method=\"jitter\",ylab=\"Normalised log2 expression\",main=my_gene)\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Activity\n",
    "Create a barplot showing  the number of up and down regulated genes in all contrasts.\n",
    "\n",
    "#### Hint\n",
    "Use the the table returned by `summary(decideTests(fit_voom2, method=\"separate\"))` as data object and either the `barplot()` function or ggplot2 `geom_bar(stat=\"identity\")` layer for build the plot."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## P-value distribution\n",
    "\n",
    "```{r}\n",
    "ggplot(Xiangfei.EL35vsItalia.EL35[!is.na(Xiangfei.EL35vsItalia.EL35$P.Value),], aes(x = P.Value)) +\n",
    "    geom_histogram(alpha=.5, position='identity', bins = 50) +\n",
    "    labs(title='Histogram of unadjusted p-values') +\n",
    "    xlab('Unadjusted p-values') +\n",
    "    xlim(c(0, 1.0005)) \n",
    "```\n",
    "\n",
    "## Volcano plot\n",
    "\n",
    "A volcano plot displays log fold changes on the x-axis versus a measure of statistical significance on the y-axis(Here $-log_{10}Pvalue$).\n",
    "\n",
    "```{r}\n",
    "results_voom <- decideTests(fit_voom2)\n",
    "summary(decideTests(fit_voom2, lfc=1, p.value=0.05))\n",
    "volcanoplot(fit_voom2, coef =1, highlight=20, names=rownames(fit_voom2), main=\"\", pch=8, cex=0.35)\n",
    "abline(h = -4.32, v = -1.5, col = \"red\", lty = 3); abline(h = 4.32, v = 1.5, col = \"red\", lty = 3)\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Heat map\n",
    "Heatmaps are a way to display differential expression results. To create a heatmap, we  select the first n DE genes log2-counts-per-million (logCPM) values.\n",
    "```{r}\n",
    "tT <- Xiangfei.EL35vsItalia.EL35\n",
    "o <- order(tT$P.Value)\n",
    "logCPM_o <- logCPM[o[1:30],]\n",
    "```\n",
    "\n",
    "A heat map can then be produced by the coolmap function which clusters genes by correlation (highly correlated genes are closest) and clusters samples based on Euclidean distance between the expression values.\n",
    "```{r}\n",
    "coolmap(logCPM_o, margins=c(7,7), lhei=c(1,6), lwid=c(1,3))\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functional Analysis (Pathway analysis)\n",
    "\n",
    "## Gene ontology analysis using GOseq \n",
    "\n",
    "GOseq allows to perform a Gene Ontology (GO) analysis tailored to RNA-seq data as it accounts for the gene length bias in detection of over-representation  <a href=\"#ref12\">[12]</a>\n",
    "\n",
    "From the [GOseq vignette](https://www.bioconductor.org/packages/release/bioc/vignettes/goseq/inst/doc/goseq.pdf):\n",
    "\n",
    "- GOseq first needs to quantify the length bias present in the dataset under consideration.\n",
    "- This is done by calculating a Probability Weighting Function or PWF which can be thought of as a function which gives the probability that a gene will be differentially expressed (DE), based on its length alone.\n",
    "- The PWF is calculated by fitting a monotonic spline to the binary data series of differential expression (1=DE, 0=Not DE) as a function of gene length.\n",
    "- The PWF is used to weight the chance of selecting each gene when forming a null distribution for GO category membership.\n",
    "- The fact that the PWF is calculated directly from the dataset under consideration makes this approach robust, only correcting for the length bias present in the data.\n",
    ">“GO analysis of RNA-seq data requires the use of random sampling in order to generate a suitable null distribution for GO category membership and calculate each category’s significance for over representation amongst DE genes. … In most cases, the Wallenius distribution can be used to approximate the true null distribution, without any significant loss in accuracy. The goseq package implements this approximation as its default option.”\n",
    "\n",
    "In order to perform the analysis two vectors are required: one containing all genes assayed in the RNA-seq experiment, the other containing all genes which are DE. If we assume that the vector of genes being assayed is namedassayed.genes and the vector of DE genes is named de.genes we can construct a named vector\n",
    "suitable for use with goseq using the following:\n",
    "\n",
    "```{r}\n",
    "library(goseq)\n",
    "assayed.genes <- fc$annotation$GeneID\n",
    "de.genes <-  Xiangfei.EL35vsXiangfei.EL36[(Xiangfei.EL35vsXiangfei.EL36$logFC > 1) & Xiangfei.EL35vsXiangfei.EL36$adj.P.Val< 0.01 ,]\n",
    "gene.vector <- as.integer(assayed.genes%in%rownames(de.genes))\n",
    "names(gene.vector) <- assayed.genes\n",
    "```\n",
    "\n",
    "### Activity\n",
    "Replace the select contrast with one of your interest end change the double cut-off for the selection of DE genes for the enrichment analysis\n",
    "\n",
    "## Non-native Gene Identifier or category test\n",
    "If the organism, Gene Identifier or category test you wish to perform is not in the native goseq\n",
    "database, you will have to supply one or all of the following:\n",
    "- Length data: the length of each gene in your gene identifier format.\n",
    "- Category mappings: the mapping (usually many-to-many) between the categories you wish to test for over/under representation amongst DE genes and genes in your gene identifier format.\n",
    "```{r}\n",
    "GO_ann=read.table(\"/home/share/RNASeqPractical/references/GO_plant_slim_fixed.tab\",header=F,sep=\"\\t\")[,1:2]\n",
    "colnames(GO_ann) <- c(\"GeneIDs\",\"GOTerms\") \n",
    "```\n",
    "\n",
    "Note: GO annotation is retrieved from [here](http://genomes.cribi.unipd.it/DATA/V1/ANNOTATION/).\n",
    "\n",
    "## GO Analysis\n",
    "\n",
    "### Fitting the Probability Weighting Function (PWF)\n",
    "We first need to obtain a weighting for each gene, depending on its length, given by the PWF.\n",
    "As you may have noticed when running supportedGenomes or supportedGeneIDs, length data is\n",
    "available in the local database for our gene ID, “ensGene” and our genome, “hg19”. We will let\n",
    "goseq automatically fetch this data from its databases.\n",
    "```{r}\n",
    "pwf=nullp(gene.vector,bias.data=fc$annotation$Length)\n",
    "```\n",
    "### Using the Wallenius approximation\n",
    "To start with we will use the default method, to calculate the over and under expressed GO\n",
    "categories among DE genes.\n",
    "```{r}\n",
    "GO.wall=goseq(pwf,gene2cat=GO_ann)\n",
    "head(GO.wall)\n",
    "```\n",
    "### Making sense of the results\n",
    "If you wish to identify categories significantly enriched/unenriched below some p-value cutoff, it is recommended to\n",
    "first apply some kind of multiple hypothesis testing correction. For example, GO categories over\n",
    "enriched using a .05 FDR cutoff are:\n",
    "```{r}\n",
    "enriched.GO=GO.wall$category[p.adjust(GO.wall$over_represented_pvalue, method=\"BH\")<.05]\n",
    "```\n",
    "Information about each term can be obtained from the [Gene Ontology website](http://www.geneontology.org) or tool like [http://revigo.irb.hr/](http://revigo.irb.hr/) or [http://bioinfo.cau.edu.cn/agriGO/](http://bioinfo.cau.edu.cn/agriGO/) or in R:\n",
    "```{r}\n",
    "library(GO.db)\n",
    "for(go in enriched.GO){\n",
    "print(GOTERM[[go]])\n",
    "cat(\"--------------------------------------\\n\")\n",
    "}\n",
    "```\n",
    "### Plot top 10 GOTERMS <a href=\"#ref13\">[13]</a> \n",
    "\n",
    "```{r}\n",
    "GO.wall %>% \n",
    "    top_n(10, wt=-over_represented_pvalue) %>% \n",
    "    mutate(hitsPerc=numDEInCat*100/numInCat) %>% \n",
    "    ggplot(aes(x=hitsPerc, \n",
    "               y= paste0(category,\" - \", term), \n",
    "               colour=over_represented_pvalue, \n",
    "               size=numDEInCat)) +\n",
    "        geom_point() +\n",
    "        expand_limits(x=0) +\n",
    "        labs(x=\"Hits (%)\", y=\"GO term\", colour=\"p value\", size=\"Count\")\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Packages used\n",
    "This practical depends on various packages from version 3.10 of the Bioconductor project, running on R version 3.6.2 or higher. The complete list of the packages used for this workflow are shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 3.6.2 (2019-12-12)\n",
       "Platform: x86_64-pc-linux-gnu (64-bit)\n",
       "Running under: Debian GNU/Linux 10 (buster)\n",
       "\n",
       "Matrix products: default\n",
       "BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.8.0\n",
       "LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.8.0\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n",
       " [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n",
       " [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n",
       "[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n",
       "\n",
       "attached base packages:\n",
       "[1] parallel  stats4    stats     graphics  grDevices utils     datasets \n",
       "[8] methods   base     \n",
       "\n",
       "other attached packages:\n",
       " [1] GO.db_3.10.0                AnnotationDbi_1.48.0       \n",
       " [3] goseq_1.38.0                geneLenDataBase_1.22.0     \n",
       " [5] BiasedUrn_1.07              edgeR_3.28.0               \n",
       " [7] limma_3.42.0                DESeq2_1.26.0              \n",
       " [9] SummarizedExperiment_1.16.1 DelayedArray_0.12.2        \n",
       "[11] BiocParallel_1.20.1         matrixStats_0.55.0         \n",
       "[13] Biobase_2.46.0              GenomicRanges_1.38.0       \n",
       "[15] GenomeInfoDb_1.22.0         IRanges_2.20.2             \n",
       "[17] S4Vectors_0.24.3            BiocGenerics_0.32.0        \n",
       "[19] tximport_1.14.0             Rsubread_2.0.1             \n",
       "[21] RColorBrewer_1.1-2          plotly_4.9.1               \n",
       "[23] forcats_0.4.0               stringr_1.4.0              \n",
       "[25] dplyr_0.8.3                 purrr_0.3.3                \n",
       "[27] readr_1.3.1                 tidyr_1.0.2                \n",
       "[29] tibble_2.1.3                ggplot2_3.2.1              \n",
       "[31] tidyverse_1.3.0            \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       "  [1] readxl_1.3.1             uuid_0.1-2               backports_1.1.5         \n",
       "  [4] Hmisc_4.3-0              BiocFileCache_1.10.2     repr_1.1.0              \n",
       "  [7] lazyeval_0.2.2           splines_3.6.2            crosstalk_1.0.0         \n",
       " [10] digest_0.6.23            htmltools_0.4.0          gdata_2.18.0            \n",
       " [13] fansi_0.4.1              magrittr_1.5             checkmate_1.9.4         \n",
       " [16] memoise_1.1.0            cluster_2.1.0            Biostrings_2.54.0       \n",
       " [19] annotate_1.64.0          modelr_0.1.5             askpass_1.1             \n",
       " [22] prettyunits_1.1.1        jpeg_0.1-8.1             colorspace_1.4-1        \n",
       " [25] blob_1.2.1               rvest_0.3.5              rappdirs_0.3.1          \n",
       " [28] haven_2.2.0              xfun_0.12                crayon_1.3.4            \n",
       " [31] RCurl_1.98-1.1           jsonlite_1.6             genefilter_1.68.0       \n",
       " [34] survival_3.1-8           glue_1.3.1               gtable_0.3.0            \n",
       " [37] zlibbioc_1.32.0          XVector_0.26.0           btools_0.0.1            \n",
       " [40] scales_1.1.0             DBI_1.1.0                Rcpp_1.0.3              \n",
       " [43] viridisLite_0.3.0        xtable_1.8-4             progress_1.2.2          \n",
       " [46] htmlTable_1.13.3         foreign_0.8-72           bit_1.1-15.1            \n",
       " [49] Formula_1.2-3            htmlwidgets_1.5.1        httr_1.4.1              \n",
       " [52] gplots_3.0.1.2           acepack_1.4.1            pkgconfig_2.0.3         \n",
       " [55] XML_3.99-0.3             farver_2.0.3             nnet_7.3-12             \n",
       " [58] dbplyr_1.4.2             locfit_1.5-9.1           later_1.0.0             \n",
       " [61] tidyselect_1.0.0         labeling_0.3             rlang_0.4.4             \n",
       " [64] munsell_0.5.0            cellranger_1.1.0         tools_3.6.2             \n",
       " [67] cli_2.0.1                generics_0.0.2           RSQLite_2.2.0           \n",
       " [70] broom_0.5.4              fastmap_1.0.1            evaluate_0.14           \n",
       " [73] yaml_2.2.0               knitr_1.27               bit64_0.9-7             \n",
       " [76] fs_1.3.1                 caTools_1.18.0           nlme_3.1-143            \n",
       " [79] mime_0.8                 xml2_1.2.2               biomaRt_2.42.0          \n",
       " [82] compiler_3.6.2           rstudioapi_0.10          curl_4.3                \n",
       " [85] png_0.1-7                reprex_0.3.0             geneplotter_1.64.0      \n",
       " [88] stringi_1.4.5            GenomicFeatures_1.38.1   lattice_0.20-38         \n",
       " [91] IRdisplay_0.7.0          Matrix_1.2-18            vctrs_0.2.2             \n",
       " [94] pillar_1.4.3             lifecycle_0.1.0          data.table_1.12.8       \n",
       " [97] bitops_1.0-6             httpuv_1.5.2             rtracklayer_1.46.0      \n",
       "[100] R6_2.4.1                 latticeExtra_0.6-29      promises_1.1.0          \n",
       "[103] KernSmooth_2.23-16       gridExtra_2.3            gtools_3.8.1            \n",
       "[106] assertthat_0.2.1         openssl_1.4.1            withr_2.1.2             \n",
       "[109] GenomicAlignments_1.22.1 Rsamtools_2.2.1          GenomeInfoDbData_1.2.2  \n",
       "[112] mgcv_1.8-31              hms_0.5.3                grid_3.6.2              \n",
       "[115] rpart_4.1-15             IRkernel_1.1             pbdZMQ_0.3-3            \n",
       "[118] shiny_1.4.0              lubridate_1.7.4          base64enc_0.1-3         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "(my_session <- sessionInfo())\n",
    "#save.image(paste0(\"subread_analysis_\",format(Sys.Date(), \"%d-%m-%Y\"),\".RData\"), compress = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# References\n",
    "\n",
    "<a id='ref1'></a>1. Sun,L., Zhu,B., Zhang,X., Zhang,G., Yan,A., Wang,H., Wang,X. and Xu,H. (2019) Transcriptome profiles of three Muscat table grape cultivars to dissect the mechanism of terpene biosynthesis. Sci Data, 6.\n",
    "\n",
    "<a id='ref2'></a>2. [Mapping between SRA IDs and accessing metadata with the Bioconductor SRAdb package](http://bergmanlab.genetics.uga.edu/?p=1072)\n",
    "\n",
    "<a id='ref2bis'></a>2bis. Choudhary, S. (2019) pysradb: A Python package to query next-generation sequencing metadata and data from NCBI Sequence Read Archive. F1000Res, 8, 532.\n",
    "\n",
    "<a id='ref3'></a>3. Babraham Bioinformatics - FastQC A Quality Control tool for High Throughput Sequence Data. Babraham Bioinformatics - FastQC A Quality Control tool for High Throughput Sequence Data \n",
    "https://www.bioinformatics.babraham.ac.uk/projects/fastqc/ (accessed Nov 21, 2019).\n",
    "\n",
    "<a id='ref4'></a>4. Bolger, A. M., Lohse, M. and Usadel, B. (2014) Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics, 30, 2114–2120.\n",
    "\n",
    "<a id='ref5'></a>5. Y. Liao, G. K. Smyth and W. Shi. “The Subread aligner: fast, accurate and scalable read mapping by seed-and-vote”. In: Nucleic Acids Research 41 (2013), p. 108.\n",
    "\n",
    "<a id='ref6'></a>6. Liao, Y., Smyth, G. K. and Shi, W. (2014) featureCounts: an efficient general purpose program for assigning sequence reads to genomic features. Bioinformatics, 30, 923–930.\n",
    "\n",
    "<a id='ref7'></a>7. [Principal Component Analysis Explained Visually](http://setosa.io/ev/principal-component-analysis/)\n",
    "\n",
    "<a id='ref8'></a>8. Robinson, M. D., and A. Oshlack. 2010. “A Scaling Normalization Method for Differential Expression Analysis of RNA-seq Data.” Genome Biology 11 (3):R25.\n",
    "\n",
    "<a id='ref9'></a>9. Chen, Y., Lun, A. T. L. and Smyth, G. K. (2016) From reads to genes to pathways: differential expression analysis of RNA-Seq experiments using Rsubread and the edgeR quasi-likelihood pipeline. F1000Research, 5, 1438.\n",
    "\n",
    "<a id='ref10'></a>10. Law, C. W., Chen, Y., Shi, W., et al. (2014) voom: precision weights unlock linear model analysis tools for RNA-seq read counts. Genome Biology, 15, R29.\n",
    "\n",
    "<a id='ref11'></a>11. Ritchie, M. E., Phipson, B., Wu, D., et al. (2015) limma powers differential expression analyses for RNA-sequencing and microarray studies. Nucl. Acids Res., gkv007.\n",
    "\n",
    "<a id='ref12'></a>12. Young, M. D., Wakefield, M. J., Smyth, G. K., et al. (2010) Gene ontology analysis for RNA-seq: accounting for selection bias. Genome Biology, 11, R14.\n",
    "\n",
    "<a id='ref13'></a>13. [GO enrichment analysis](https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2018/RNASeq2018/html/06_Gene_set_testing.nb.html#go-enrichment-analysis)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
